pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -419,33 +419,41 @@ func generateAltVirtualHosts(hostname string, port int, proxyDomain string) []st
 		return nil
 	}
 
+	// For Kubernetes services, If Kubernetes MCS host is enabled, also add a virtual host for 'clusterset.local'
+	if features.EnableMCSHost {
+		svcIndex := strings.LastIndex(hostname, ".svc.")
```

Comment:
So even if the `proxyDomain` is not `*.cluster.local` we will do this?

Reply:
yeah that's right ... even if there is a custom domain, we'll still create a `clusterset.local` host.

Reply:
I think this check is not fully safe. Probably fine for 99% of cases, but you could have `foo.svc.google.com`. It should only match if its `.svc.<cluster-domain>`. It looks like we are checking that a bit below?